.github/workflows/htmcore.yml

```diff
@@ -0,0 +1,157 @@
+name: htm.core build
+
+on:
+  push:
+    branches:
+      - v*\.*\.*\.* # vX.Y.Z release tag
+  # run on pull_request events that target the master branch
+  pull_request:
+    branches:
+    - master
+  # run every day of the week from Monday - Friday at 02:00
+  schedule:
+  - cron: 0 2 * * 1-5
+      
+jobs:
+  build:
+    name: Building on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      #max-parallel: 4
+      matrix:
+        python-version: [3.7]
+        os: [ubuntu-18.04, windows-2019, macOS-latest]
+
+    steps:
+    - uses: actions/checkout@v1
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Versions
+      run: |
+        python --version
+        cmake --version
+        c++ --version
+
+    - name: Install dependencies (gcc-8)
+      if: matrix.os == 'ubuntu-18.04'
+      env:
+        CC: gcc-8
```

Comment:
setting custom gcc (gcc-8, so we use c++17 and not :boost) is quite challenging here